### PR TITLE
Implement Query Log sorting (server-side pagination)

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

The PR adds the ability to sort the server-side prepared Query Log by all relevant columns. This PR works on its own, no `web` changes are required to get Query Log sorting.

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: src/api/queries.c [hunk 1]
```diff
@@ -338,14 +383,33 @@ int api_queries(struct ftl_conn *api)
...
+		{
+			log_debug(DEBUG_API, "Sorting by column %d (%s)", sort_column, sort_dir);
+
+			char sort_col_id[32] = { 0 };
+			snprintf(sort_col_id, sizeof(sort_col_id), "columns%%5B%d%%5D%%5Bdata%%5D",
+			         sort_column);
+
+			// Encoded URI string: %5B = [ and %5D = ]
+			if(GET_VAR(sort_col_id, sort_col, api->request->query_string) > 0)
+				log_info("Sorting by column %s (%s)", sort_col, sort_dir);
```
Comment (yubiuser) [inline src/api/queries.c:401]:
```suggestion
				log_debug("Sorting by column %s (%s)", sort_col, sort_dir);
```

Info is to much....

```

nanopi@nanopi:/var/www/html/admin$ tail -f /var/log/pihole/FTL.log 
2023-10-12 22:48:22.697 [52333/T52344] INFO: Sorting by column client.ip (desc)
2023-10-12 22:48:36.943 [52333/T52341] INFO: Sorting by column client.ip (desc)
2023-10-12 22:48:39.229 [52333/T52342] INFO: Sorting by column domain (asc)
2023-10-12 22:48:41.063 [52333/T52343] INFO: Sorting by column client.ip (asc)
2023-10-12 22:48:42.033 [52333/T52344] INFO: Sorting by column domain (asc)
2023-10-12 22:48:43.023 [52333/T52345] INFO: Sorting by column client.ip (asc)
2023-10-12 22:48:44.117 [52333/T52346] INFO: Sorting by column status (asc)
2023-10-12 22:48:47.195 [52333/T52340] INFO: Sorting by column status (desc)
2023-10-12 22:50:06.403 [52333/T52341] INFO: Sorting by column status (desc)
2023-10-12 22:50:10.935 [52333/T52344] INFO: Sorting by column status (desc)


```

**Review by yubiuser — COMMENTED** [review]
You added way more options here

```
		if(strcasecmp(sort_col, "time") == 0)
			sort_col_sql = "timestamp";
		else if(strcasecmp(sort_col, "domain") == 0)
			sort_col_sql = "d.domain";
		else if(strcasecmp(sort_col, "client.ip") == 0)
			sort_col_sql = "c.ip";
		else if(strcasecmp(sort_col, "client.name") == 0)
			sort_col_sql = "c.name";
		else if(strcasecmp(sort_col, "upstream") == 0)
			sort_col_sql = "f.forward";
		else if(strcasecmp(sort_col, "type") == 0)
			sort_col_sql = "q.type";
		else if(strcasecmp(sort_col, "status") == 0)
			sort_col_sql = "q.status";
		else if(strcasecmp(sort_col, "reply") == 0)
			sort_col_sql = "q.reply_type";
		else if(strcasecmp(sort_col, "dnssec") == 0)
			sort_col_sql = "q.dnssec";
		else if(strcasecmp(sort_col, "regex.id") == 0)
			sort_col_sql = "regex_id";
			
```

Than what is possible to filter from the web interface. Is this on purpose?

File: src/api/queries.c [hunk 2]
```diff
@@ -398,7 +398,7 @@ int api_queries(struct ftl_conn *api)
 
 			// Encoded URI string: %5B = [ and %5D = ]
 			if(GET_VAR(sort_col_id, sort_col, api->request->query_string) > 0)
-				log_info("Sorting by column %s (%s)", sort_col, sort_dir);
+				log_debug("Sorting by column %s (%s)", sort_col, sort_dir);
```
[the remark this answers is not in the document]
Comment (DL6ER) [reply src/api/queries.c]: Fixed.

**Comment by DL6ER** [comment]
> You added way more options here

Yes. This is for future extensibility. If we ever decide to add more columns to the Query Log (I have seen at least one for adding back the Reply time as direct column on Discourse), we won't need a FTL PR. A web PR will be sufficient.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**